metaseq/checkpoint_utils.py

```diff
@@ -229,42 +230,62 @@ def load_checkpoint(cfg: CheckpointConfig, trainer, **passthrough_args):
 
     # TODO(susanz): fix all of this spagetti, split out logic by env
     # Note that we compare by value since ComputeEnvs may be imported from metaseq_internal
-    if (
-        cfg.cloud_upload_path
-        and cfg.cluster_env == ComputeEnvs.AZURE.value
-        and has_metaseq_internal
-    ):
-        if (
-            # --restore-file was not passed, always download latest checkpoint
-            (
-                cfg.restore_file == default_restore_file
-                and cfg.finetune_from_model is None
-            )
-            # --restore-file was passed, but we requeued, so download latest checkpoint
-            or int(os.environ.get("SLURM_RESTART_COUNT", 0)) > 0
-        ):
-            # download checkpoint into local save_dir
-            checkpoint_path_to_load = os.path.join(
-                cfg.save_dir, "checkpoint_last{}.pt".format(suffix)
-            )
-            azure_utils.download_recent_ckpt(
-                cfg.cloud_upload_path, checkpoint_path_to_load, suffix + ".pt"
-            )
-        elif (
-            # --restore-file was passed and is a blob URL, download that checkpoint
-            cfg.restore_file != default_restore_file
-            and "windows.net" in cfg.restore_file
-        ):
-            blob_url = cfg.restore_file.replace(".pt", suffix + ".pt")
-            # download checkpoint into local save_dir
+
+    if cfg.cloud_upload_path:
+        if cfg.cloud_upload_path.startswith("nfs:"):
             checkpoint_path_to_load = os.path.join(
                 cfg.save_dir, "checkpoint_last{}.pt".format(suffix)
             )
-            azure_utils.download_specific_ckpt(blob_url, checkpoint_path_to_load)
-        else:
-            logger.info(
-                f"Using checkpoint {checkpoint_path_to_load} even while on Azure"
-            )
+            nfs_path = cfg.cloud_upload_path[4:]
+            filename = None
+            if cfg.restore_file == default_restore_file or int(os.environ.get("SLURM_RESTART_COUNT", 0)) > 0:
```

Comment:
Could we encapsulate the three checks in a separate method? We should also check for `finetune_from_model` here too, no?

`if (
                # --restore-file was not passed, always download latest checkpoint
                (
                    cfg.restore_file == default_restore_file
                    and cfg.finetune_from_model is None
                )
                # --restore-file was passed, but we requeued, so download latest checkpoint
                or int(os.environ.get("SLURM_RESTART_COUNT", 0)) > 0
            ):`